### PR TITLE
DBZ-1024 Emit correct last_snapshot_record in source

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-1024

### CODE REVIEW

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsSnapshotProducer.java [hunk 1]
```diff
@@ -203,6 +203,7 @@ private void takeSnapshot(BlockingConsumer<ChangeEvent> consumer) {
                 // process and send the last record after marking it as such
                 logger.info("Step 5: sending the last snapshot record");
                 sourceInfo.markLastSnapshotRecord();
+                changeSourceToLastSnapshotRecord(currentRecord);
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsSnapshotProducer.java]: Hum, something is weird here: if I comment out this line, still all tests that make use of `assertRecordOffsetAndSnapshotSource()` pass. Shouldn't any of them fail? Also, why isn't this part of `markLastSnapshotRecord()` in the line above?

Comment (jpechane) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsSnapshotProducer.java]: @gunnarmorling Thanks for spotting, I inadverently removed one assert when I was copying around the code to handle heartbeat events.

Comment (gunnarmorling) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsSnapshotProducer.java]: Ok, got it. It feels a bit convoluted, but I can't think of a really better way right now either.

**Comment by gunnarmorling** [comment]
Rebased and applied. Added a commit for some small clean-up, too. Thanks, @jpechane!

**Pull request closed without merging**